src/main/java/sim/coffee/Server.java

```diff
@@ -0,0 +1,26 @@
+package sim.coffee;
+
+public class Server implements Runnable{
+    private Queue queue;
+
+    public Server(Queue queue) {
+        this.queue = queue;
+    }
+
+    public void run() {
+        while (queue.getCustomerCount() == 0) {
+            try {
+                Thread.sleep(100);
+            } catch (InterruptedException e) {}
+        }
+        queue.getCustomerOrder();
+        if (queue.getDone()) {
+            try {
+                Thread.sleep(100);
+            } catch (InterruptedException e) {}
+        } else {
+            queue.setDone();
+        }
```

Comment:
I think Lina is right that this if statement should instead just terminate the server thread (because all orders are done)

Reply:
Got it, looking at it right now